faction.cpp

```diff
@@ -559,34 +790,27 @@ void Faction::WriteReport(ostream& f, Game *pGame, int ** citems)
 		f << '\n';
 	}
 
-	if (shows.Num()) {
+	if (!shows.empty()) {
```

Comment:
I converted shows/objectshows/itemsshows into std::vector<> rather than Alists..  Slowly but surely, AList and AString will die too!